include/eve/algo/all_of.hpp

```diff
@@ -7,60 +7,95 @@
 #include <eve/algo/concepts.hpp>
 #include <eve/algo/for_each_iteration.hpp>
 #include <eve/algo/preprocess_range.hpp>
 #include <eve/algo/traits.hpp>
-
+#include <eve/module/core.hpp>
 
 #include <array>
 #include <utility>
 
 namespace eve::algo
 {
-  template <typename TraitsSupport>
-  struct all_of_ : TraitsSupport
+template<typename TraitsSupport> struct all_of_ : TraitsSupport
+{
+  template<typename P> struct delegate
   {
-    template <typename P>
-    struct delegate
-    {
-      explicit delegate(P p) : p(p) {}
+    explicit delegate(P p) : p(p) {}
 
-      EVE_FORCEINLINE bool step(auto it, eve::relative_conditional_expr auto ignore, auto /*idx*/)
-      {
-        eve::logical test = p(eve::load[ignore](it));
-        res = eve::all[ignore](test);
-        return !res;
-      }
+    EVE_FORCEINLINE bool step(auto it, eve::relative_conditional_expr auto ignore, auto /*idx*/)
+    {
+      eve::logical test = p(eve::load[ignore](it));
+      res               = eve::all[ignore](test);
+      return !res;
+    }
 
-      template <typename I, std::size_t size>
-      EVE_FORCEINLINE bool unrolled_step(std::array<I, size> arr)
-      {
-        auto tests = array_map(arr, load_and_apply{p});
-        auto overall = array_reduce(tests, eve::logical_and);
-        res = eve::all(overall);
-        return !res;
-      }
+    template<typename I, std::size_t size>
+    EVE_FORCEINLINE bool unrolled_step(std::array<I, size> arr)
+    {
+      auto tests   = array_map(arr, load_and_apply {p});
+      auto overall = array_reduce(tests, eve::logical_and);
+      res          = eve::all(overall);
+      return !res;
+    }
 
-      P p;
-      bool res = true;
-    };
+    P    p;
+    bool res = true;
+  };
 
-    template <relaxed_range Rng, typename P>
-    EVE_FORCEINLINE bool operator()(Rng&& rng, P p) const
-    {
-      auto processed = preprocess_range(TraitsSupport::get_traits(), EVE_FWD(rng));
+  template<relaxed_range Rng, typename P> EVE_FORCEINLINE bool operator()(Rng&& rng, P p) const
+  {
+    auto processed = preprocess_range(TraitsSupport::get_traits(), EVE_FWD(rng));
 
-      if (processed.begin() == processed.end()) return true;
+    if( processed.begin() == processed.end() ) return true;
 
-      delegate<P> d{p};
-      algo::for_each_iteration(processed.traits(), processed.begin(), processed.end())(d);
-      return d.res;
-    }
-  };
+    delegate<P> d {p};
+    algo::for_each_iteration(processed.traits(), processed.begin(), processed.end())(d);
+    return d.res;
+  }
+};
 
-  inline constexpr auto all_of = function_with_traits<all_of_>[default_simple_algo_traits];
+//================================================================================================
+//! @addtogroup algos
+//! @{
+//!   @var all_of
+//!
+//!   @brief a SIMD version of `std::all_of`
+//!
+//!   Tests if all of the elements in the range match the predicate.
+//!   by default aligns and unrolls 4 times.
+//!
+//!   @code
```

Comment:
<valid for all files>
We usually put `**Alternative Header**` just before the algo.hpp code block